storage_v2/converter/factory.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package converter
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+	storage_v1 "github.com/jaegertracing/jaeger/storage"
+	dependencystore_v1 "github.com/jaegertracing/jaeger/storage/dependencystore"
+	spanstore_v1 "github.com/jaegertracing/jaeger/storage/spanstore"
+	"github.com/jaegertracing/jaeger/storage_v2/spanstore"
+)
+
+type Factory struct {
+	ss spanstore.Factory
+}
+
+func NewFactory(ss spanstore.Factory) storage_v1.Factory {
```

Comment:
please add explanation/docs what this is for

Reply:
How do you see this being used? It seems to be going in the opposite direction.

The way I imagined we would proceed is we have `jaegerexporter` that currently receives OTLP but then has to translate it into `[]model.Span` and write to storage_v1. The next step would be to make `jaegerexporter` only operate on OTLP by writing to storage_v2 API. Then this adapter comes in handy because when we as `jaegerstorage` extension to GetFactory(name), it can return locate storage_v1 factory and wrap it in the adapter to make it look like storage_v2.

Reply:
Interesting, seems like your approach is more precise.

I was thinking that with this changes we can blend in the storage_v2 to `jaegerstorage` extension so that it can support both v1 & v2 while we're implementing the storage_v2 plugin gradually. After all the storage_v2 plugins have been implemented, then we can change the `storageexporter` and `storagequery` to use storage_v2 API directly and remove the converter.

I like your approach more, will change that.